services/skus/service.go

```diff
@@ -1329,129 +1328,151 @@ func (s *Service) verifyCredential(ctx context.Context, req credential, w http.R
+	logger.Debug().Str("presentation", string(bytes)).Msg("presentation unmarshalled")
+
+	// Ensure that the credential being redeemed (opaque to merchant) matches the outer credential details
+	issuerID, err := encodeIssuerID(req.GetMerchantID(ctx), req.GetSku(ctx))
+	if err != nil {
+		logger.Error().Err(err).
+			Msg("failed to encode the issuer id")
+		return handlers.WrapError(err, "Error in outer merchantId or sku", http.StatusBadRequest)
+	}
+	logger.Debug().Str("issuer", issuerID).Msg("issuer encoded")
+
+	keys, err := s.GetCredentialSigningKeys(ctx, req.GetMerchantID(ctx))
+	if err != nil {
+		return handlers.WrapError(err, "failed to get merchant signing key", http.StatusInternalServerError)
+	}
+
+	issuedAt, err := time.Parse("2006-01-02", presentation.IssuedAt)
+	if err != nil {
+		logger.Error().Err(err).
+			Msg("failed to parse issued at time of credential")
+		return handlers.WrapError(err, "Error parsing issuedAt", http.StatusBadRequest)
+	}
+	expiresAt, err := time.Parse("2006-01-02", presentation.ExpiresAt)
+	if err != nil {
+		logger.Error().Err(err).
+			Msg("failed to parse expires at time of credential")
+		return handlers.WrapError(err, "Error parsing expiresAt", http.StatusBadRequest)
+	}
+
+	for _, key := range keys {
+		timeLimitedSecret := cryptography.NewTimeLimitedSecret(key)
+		verified, err := timeLimitedSecret.Verify([]byte(issuerID), issuedAt, expiresAt, presentation.Token)
 		if err != nil {
 			logger.Error().Err(err).
-				Msg("failed to parse expires at time of credential")
-			return handlers.WrapError(err, "Error parsing expiresAt", http.StatusBadRequest)
+				Msg("failed to verify time limited credential")
+			return handlers.WrapError(err, "Error in token verification", http.StatusBadRequest)
 		}
 
-		for _, key := range keys {
-			timeLimitedSecret := cryptography.NewTimeLimitedSecret(key)
-			verified, err := timeLimitedSecret.Verify([]byte(issuerID), issuedAt, expiresAt, presentation.Token)
-			if err != nil {
-				logger.Error().Err(err).
-					Msg("failed to verify time limited credential")
-				return handlers.WrapError(err, "Error in token verification", http.StatusBadRequest)
-			}
-
-			if verified {
-				// check against expiration time, issued time
-				if time.Now().After(expiresAt) || time.Now().Before(issuedAt) {
-					logger.Error().
-						Msg("credentials are not valid")
-					return handlers.RenderContent(ctx, "Credentials are not valid", w, http.StatusForbidden)
-				}
-				logger.Debug().Msg("credentials verified")
-				return handlers.RenderContent(ctx, "Credentials successfully verified", w, http.StatusOK)
+		if verified {
+			// check against expiration time, issued time
+			if time.Now().After(expiresAt) || time.Now().Before(issuedAt) {
+				logger.Error().
+					Msg("credentials are not valid")
+				return handlers.WrapError(nil, "Credentials are not valid", http.StatusForbidden)
 			}
+			logger.Debug().Msg("credentials verified")
```

Comment:
same as [can we remove debug logging](https://github.com/brave-intl/bat-go/pull/1995/files#r1331720525)